cmd/controller/main.go

```diff
@@ -27,20 +27,15 @@ import (
 )
 
 func main() {
-	options, manager := operator.NewOptionsWithManagerOrDie()
-	cloudProvider := cloudprovider.CloudProvider(awscloudprovider.NewCloudProvider(options.Ctx, cloudprovider.Options{
-		ClientSet:  options.Clientset,
-		KubeClient: options.KubeClient,
-		StartAsync: options.StartAsync,
-	}))
-	if hp, ok := cloudProvider.(operator.HealthCheck); ok {
-		utilruntime.Must(manager.AddHealthzCheck("cloud-provider", hp.LivenessProbe))
-	}
-	cloudProvider = cloudprovidermetrics.Decorate(cloudProvider)
-	if err := operator.RegisterControllers(options.Ctx,
-		manager,
-		controllers.GetControllers(options, cloudProvider)...,
-	).Start(options.Ctx); err != nil {
+	ctx := operator.NewContextOrDie()
+	cloudProvider := awscloudprovider.NewCloudProvider(ctx, cloudprovider.Options{
```

Comment:
I have some small concerns about exposing the manager through the context in this way. If someone does a typed conversion to the `operator.Context` object within the cloudprovider, they could then have access to the manager which seems like something we want to avoid

Reply:
Agree. Will play with this.